loader.js

```diff
@@ -8,9 +8,13 @@ var NodeTargetPlugin = require("webpack/lib/node/NodeTargetPlugin");
 var LibraryTemplatePlugin = require("webpack/lib/LibraryTemplatePlugin");
 var SingleEntryPlugin = require("webpack/lib/SingleEntryPlugin");
 var LimitChunkCountPlugin = require("webpack/lib/optimize/LimitChunkCountPlugin");
+
+var extractTextPluginHmrRuntime = require.resolve("./hotModuleReplacement.js");
+
 module.exports = function(source) {
 	if(this.cacheable) this.cacheable();
-	return source;
+	// Even though this gets overwritten if extract+remove are true, without it, the runtime doesn't get added to the chunk
+	return "if(module.hot){require('" + extractTextPluginHmrRuntime + "');}\n" + source;
```

Comment:
Match the current style for now. We have a stronger solution (**webpack-defaults**) in the pipeline that makes it easier to handle.